manim/mobject/types/vectorized_mobject.py

```diff
@@ -983,13 +988,15 @@ def point_from_proportion(self, alpha: float) -> np.ndarray:
             Point evaluated.
         """
 
-        curves_with_lengths = list(self.get_curve_functions_with_lengths())
+        if alpha == 1:
+            return self.points[-1]
 
-        total_length = np.sum(length for _, length in curves_with_lengths)
-        target_length = alpha * total_length
+        curves_and_lengths = tuple(self.get_curve_functions_with_lengths())
+
+        target_length = alpha * np.sum([length for _, length in curves_and_lengths])
```

Comment:
You don't need to make a new list
```suggestion
        target_length = alpha * np.sum(length for _, length in curves_and_lengths)
```